dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/StaticLinkageChecker.java

```diff
@@ -245,6 +254,38 @@ JarLinkageReport generateLinkageReport(Path jarPath, SymbolReferenceSet symbolRe
     return Optional.of(LinkageErrorMissingMethod.errorAt(reference));
   }
 
+  /**
+   * Returns an {@code Optional} describing the linkage error for the field reference if the
+   * reference does not have a valid referent in the input class path; otherwise an empty {@code
+   * Optional}.
+   */
+  private Optional<LinkageErrorMissingField> checkLinkageErrorAt(FieldSymbolReference reference) {
+    String targetClassName = reference.getTargetClassName();
+    String fieldName = reference.getFieldName();
+    try {
+      JavaClass javaClass = classDumper.loadJavaClass(targetClassName);
+      while (javaClass != null) {
```

Comment:
I feel like there must be a way to extract out common logic from this and `validateMethodReferenceByBcelRepository` below. In both you traverse the superclasses looking for a match. They differ only in how you find a match.